mix_test.go

```diff
@@ -162,11 +164,23 @@ func newTestRoute(numHops int) ([]*SphinxNode, *SphinxPacket, error) {
 }
 
 func TestSphinxNodeReplay(t *testing.T) {
-	// We'd like to ensure that the sphinx node itself rejects all replayed
-	// packets which share the same shared secret.
-	nodes, fwdMsg, err := newTestRoute(NumMaxHops)
+	nodeKeys, nodes, route := generateRoute()
+	nodeMap := make(map[[16]byte]*SphinxNode)
+	for i := range nodes {
+		nodeMap[nodes[i].id] = nodes[i]
+	}
+	pki := NewDummyPKI(nodeKeys)
+	randReader, err := NewFixedNoiseReader("b5451d2eb2faf3f84bc4778ace6516e73e9da6c597e6f96f7e63c7ca6c9456018be9fd84883e4469a736c66fcaeceacf080fb06bc45859796707548c356c462594d1418b5349daf8fffe21a67affec10c0a2e3639c5bd9e8a9ddde5caf2e1db802995f54beae23305f2241c6517d301808c0946d5895bfd0d4b53d8ab2760e4ec8d4b2309eec239eedbab2c6ae532da37f3b633e256c6b551ed76321cc1f301d74a0a8a0673ea7e489e984543ca05fe0ff373a6f3ed4eeeaafd18292e3b182c25216aeb8")
```

Comment:
ah ok, the mix client uses a entropy source for key generation and client id and message id generation. for tests i sometimes what to test those same code paths but i want it to be deterministic, so that's what the randReader thing is for.